### DISCUSSION TITLE
Getting loss value for testing set

### DISCUSSION THREAD

**Discussion by VittoriaOssanna** [MilesCranmer/PySR#799] [Q&A] [answered]
Hello everyone, 

I am having troubles getting the loss from a test set. My dataset is divided into train and test through `sklearn`. I understand that I can use `model.predict(test_set)`, but I was wondering if there is any direct way of getting the **loss** of the test batch along with every prediction. I implemented it myself but I am sure enough that is not the smartest way. 

Can somebody help me?
Thank you!

**Comment by MilesCranmer** [comment]
Maybe this?

```python
from sklearn.metrics import mean_squared_error as mse

y_pred_test = model.predict(x_test)
mse(y_pred_test, y_test)
```

**Reply by VittoriaOssanna** [reply]
What if I used `LogitDistLoss`, `L2DIstLoss` and `L1DitstLoss`? Is there any quick way to compute the loss?

**Reply by MilesCranmer** [reply, marked as answer]
Hm, maybe try 

```python
summed_loss = jl.seval(
"""
(pred, target) -> begin
    loss = L2DistLoss()
    sum(loss(pred[i], target[i]) for i in eachindex(pred, target))
end
"""
)
summed_loss(y_pred_test, y_test)
```
